### DISCUSSION TITLE
v4.2: Beta Testing Invitation

### DISCUSSION THREAD

**Discussion by edwardtfn** [Blackymas/NSPanel_HA_Blueprint#1556] [General]
Hello NSPanel Community,

We're delighted to announce that version 4.2 of NSPanel HA Blueprint is ready for beta testing. This update introduces a series of enhancements and new features to enrich your NSPanel experience.

We invite you to participate in the beta testing, ideally on a non-production panel. Your feedback is crucial in identifying any potential issues and ensuring the stability of this release.

**What’s New in v4.2:**
- UI improvements, including additional custom buttons on the Home page and standardized entity icons.
- Introduction of features like multiple Alarm Control Panels support, dual thermostat controller support (add-on), and enhanced TFT transfer.
- Important updates such as the reversion to "Toggle" for automation entities and setting ESPHome v2023.12.0 as the minimum required version.

For a detailed overview of all changes, including breaking changes, please visit the [Release Notes](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/beta/ReleaseNotes.md).

**How to Join the Beta Testing:**
All necessary files and instructions for the beta version are available in the [`beta`](https://github.com/Blackymas/NSPanel_HA_Blueprint/tree/beta) branch of our repository. [Click here for direct access](https://github.com/Blackymas/NSPanel_HA_Blueprint/tree/beta).

**We Value Your Input:**
- If you encounter any issues during beta testing, please report them on our [Issues page](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues). Your reports are crucial for identifying and addressing problems effectively.
- If you're able to contribute fixes or improvements, feel free to create a Pull Request in the [`dev`](https://github.com/Blackymas/NSPanel_HA_Blueprint/tree/dev) branch. We greatly appreciate contributions from our community.
- For new feature ideas or enhancements, create a ["Feature Request"](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/new). Your suggestions are valuable for shaping future releases.

Every piece of feedback matters, from typos in the documentation to minor bugs. Let us know if everything looks good or if there are areas that need attention.

Thank you for your continued support and contribution to making NSPanel HA Blueprint a success. Your involvement is key to improving each version.

Happy testing, and we look forward to your valuable feedback!

**Comment by hapklaar** [comment]
I'd like to participate. Can you specify how to update to this beta? What I did doesn't seem to work:

1. Imported blueprint with url https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/beta/nspanel_blueprint.yaml and did override ==> Automation blueprint now shows Version: v4.2beta.1
2. Did a validate on the device in ESPHome, then did install
3. Did a TFT update

The panel now sit at the boot screen and seems to still be at version 4.1.4 for TFT and ESPHome, and is missing the blueprint. Log shows:

```
[14:09:34][D][script.check_versions:2175]: ESPHome version: 4.1.4
[14:09:34][D][script.check_versions:2176]: TFT version: 4.1.4
[14:09:34][D][script.check_versions:2178]: Blueprint version: 
[14:09:34][E][script.check_versions:2179]: Blueprint version mismatch!
```

**Reply by edwardtfn** [reply]
On your panel's esphome yaml, on the `remote_packages` area, change from `ref: main` to `ref: beta`. Then install it in your panel (wirelessly is fine).

**Reply by edwardtfn** [reply]
For TFT, please go to your device's page (Settings > Devices and Services > ESPHome), on the configuration area, expand it to se hidden entities and enable the "Upload TFT display - Branch" and then select "beta". Please don't forget to change this back to "main" after your tests.

**Comment by hapklaar** [comment]
> On your panel's esphome yaml, on the `remote_packages` area, change from `ref: main` to `ref: beta`. Then install it in your panel (wirelessly is fine).

~~In addition to my step 1 or instead of?~~

Thanks. It now works as it should. If others also need guidance, these are the steps I took:

1. Imported blueprint with url https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/beta/nspanel_blueprint.yaml and did override ==> Automation blueprint now shows Version: v4.2beta.1
2. On your panel's `esphome yaml`, on the `remote_packages` area, change from `ref: main` to `ref: beta`. Then install it in your panel (wirelessly is fine).
3. Go to your device in Settings > Devices and Services > ESPHome, click on `+n entity not shown` in Configuration area. Click `Update TFT display - Branch` then cogwheel and set it to Enabled.
4. Select branch beta
5. Make sure `Update TFT display - Model` is set to `Use nextion_update_url`, then update TFT display from there. I had it set to `NSPanel EU`, then it will not update to the beta, but reinstall 4.1.4.
6. Your device should update TFT, then reboot and show `4.2beta.1` for all 3 categories.

**Reply by edwardtfn** [reply]
In addition

**Comment by Floppe** [comment]
One thing does not work for me. If it is NSPanel or the Thermia integration's fault I do not know. Setting min and max temp for my Thermia tap water integration.
```

Logger: homeassistant.helpers.entity
Source: helpers/entity.py:698
First occurred: 14:33:09 (14 occurrences)
Last logged: 14:34:15

Update for climate.tap_water fails
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 698, in async_update_ha_state
    await self.async_device_update()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 959, in async_device_update
    await self.async_update()
  File "/config/custom_components/thermiagenesis/climate.py", line 234, in async_update
    await self.coordinator.wantsRefresh(self.kind)
  File "/config/custom_components/thermiagenesis/__init__.py", line 130, in wantsRefresh
    await self.coordinator.async_request_refresh()
          ^^^^^^^^^^^^^^^^
AttributeError: 'ThermiaGenesisDataUpdateCoordinator' object has no attribute 'coordinator'

````

Otherwise, wow. Great work!

**Reply by Floppe** [reply]
Side note and small UI issue: When changing min/max temps, then celsius icon is removed

**Reply by edwardtfn** [reply]
> One thing does not work for me. If it is NSPanel or the Thermia integration's fault I do not know. Setting min and max temp for my Thermia tap water integration.

Could you please try it again with beta 4 and let me know if this issue persists?

**Reply by Floppe** [reply]
Well, now it works.. although the error is still logged.

The Celsius thing could be because the Thermia integration is so slow. It takes 30+ seconds before it changes the sepoint.

**Reply by edwardtfn** [reply]
> The Celsius thing could be because the Thermia integration is so slow. It takes 30+ seconds before it changes the sepoint.

This is probably the case... I will take a look at that for v4.3. 😉

**Reply by edwardtfn** [reply]
> Side note and small UI issue: When changing min/max temps, then celsius icon is removed

- #1569

**Comment by kroessig** [comment]
hey, 
good job. Thank you very much.

One question: is it intended that a "No" is displayed for entities without a unit? It's not a bug, rather than a question. "no --> no icon there"

a wish perhaps for 4.3:
Unfortunately I can only specify sensors on the home-/climatepage not numbers. So I created a template sensor as an interim solution depented an the number_value.

**Reply by edwardtfn** [reply]
> One question: is it intended that a "No" is displayed for entities without a unit? It's not a bug, rather than a question. "no --> no icon there"

No, this is not intended. Could you please send a picture so I can understand what is happening?

- #1576

**Reply by edwardtfn** [reply]
> a wish perhaps for 4.3:
Unfortunately I can only specify sensors on the home-/climatepage not numbers. So I created a template sensor as an interim solution depented an the number_value.

- #1577

**Reply by kroessig** [reply]
![image](https://github.com/Blackymas/NSPanel_HA_Blueprint/assets/139384422/670d5812-7a12-4784-bb45-ff629586f230)
I expressed myself incorrectly. I just meant the icon and it's not "no" , it's "Non".

My templatesensor is defined as:
```
#Sensor fur NSPanel --> Heiz-Ziel Korrektur            
      - name: "Heizung_Ziel_Korrektur"
        unique_id: Heizung_Ziel_Korrektur
        state: >-
            {{ float(states('number.luxtronik2_heating_target_correction') | float(0))  }}  
```
Ah, you may fixed it already. Unfortunately I can only test it this evening.

**Reply by edwardtfn** [reply]
No, my fix was in the icon side, not in the unit if measurement... I will have to take a look again.

**Reply by edwardtfn** [reply]
Now I believe it is fixed. 😉

**Comment by lordhelmet77** [comment]
Installation of the beta went seemingly fine. The new options for text- and icon-sizes are a great and welcome addition!

So far, I see one issue: I can turn entities on, but not off. Also, long press for ie. dimmer screen does not work. Tried it on two devices, same result. Any troubleshooting ideas?

**Reply by lordhelmet77** [reply]
I deleted the blueprints and started with new ones. The issue seems to be related to the entity type:
- a Philips Hue Lightstrip works as expected
- any other lights, dimmable or not, don't
🧐

**Reply by edwardtfn** [reply]
Are you talking about button pages or some other place?

**Reply by edwardtfn** [reply]
And can you see any error message related to this on Home Assistant logs (Settings > System > Logs)?

**Reply by edwardtfn** [reply]
- #1528
- #1584 
- #1585

**Reply by lordhelmet77** [reply]
The place doesn't matter. It happens with physical buttons as well as buttons on home- and button screens.
The System Log has no entry. But the nspanel/esp32-log.
It seems that the on/off commands are fired simultaneously:

```
[23:35:36][D][binary_sensor:036]: 'nsbedroom Right Button': Sending state ON
[23:35:36][D][binary_sensor:036]: 'nsbedroom Right Button': Sending state OFF
[23:35:36][D][main:1039]: Right button - Short click
```

**Reply by lordhelmet77** [reply]
For the Hue Lightstrip, too. Maybe it is just so slow, that it doesn't care:

```
[23:38:30][D][binary_sensor:036]: 'nsbedroom Left Button': Sending state ON
[23:38:31][D][binary_sensor:036]: 'nsbedroom Left Button': Sending state OFF
[23:38:31][D][main:987]: Left button - Short click
```

**Reply by lordhelmet77** [reply]
During boot, I'll also get this error in the ESP log:

```
[23:55:40][W][nextion:405]: Nextion reported variable name invalid!
[23:55:40][E][script.set_component_color:2611]:   Component:       home.bt_notific
[23:55:40][E][script.set_component_color:2612]:   Foreground size: 0
```

**Reply by lordhelmet77** [reply]
I just found out:
- switch-entities work as expected (ie. Nspanel relays)
-  the same entity "changed" to a switch-type does only turn on
- when the buttons are in fallback-mode, the relays will only turn on, not off
- the ESP log shows that on+off are fired at the same time on each button press

**Reply by edwardtfn** [reply]
Those logs are from ESPHome. Can't you see any error on Home Assistant logs (Settings > System > Logs)?

**Reply by edwardtfn** [reply]
And on the boot screen (in your panel), which versions are shown for TFT, ESPHome and Blueprint?

**Reply by edwardtfn** [reply]
> During boot, I'll also get this error in the ESP log:
> 
> ```
> [23:55:40][W][nextion:405]: Nextion reported variable name invalid!
> [23:55:40][E][script.set_component_color:2611]:   Component:       home.bt_notific
> [23:55:40][E][script.set_component_color:2612]:   Foreground size: 0
> ```

- #1528

**Reply by lordhelmet77** [reply]
Now I see errors. These two repeat when I press a button that has a light:

```
Error while executing automation automation.nspanel_bedroom: UndefinedError: homeassistant.helpers.template.Wrapper object has no element 1
00:40:49 – (ERROR) Automation - message first occurred at 00:38:24 and shows up 4 times
NSPanel Bedroom: Main choices: NSPanel event: Choose at step 2: Button click: Choose at step 1: Hardware buttons: Error executing script. Error rendering template for variables at pos 3: UndefinedError: homeassistant.helpers.template.Wrapper object has no element 1
00:40:49 – (ERROR) Automation - message first occurred at 00:38:24 and shows up 16 times
```

The helper devices (change type to light) work normal and without errors in the log.

**Reply by lordhelmet77** [reply]
> And on the boot screen (in your panel), which versions are shown for TFT, ESPHome and Blueprint?

All three are shown as 4.2beta6

**Reply by edwardtfn** [reply]
Could you please share the attributes of this entity? You can get that from Developer Tools, tab States.

**Reply by lordhelmet77** [reply]
> Could you please share the attributes of this entity? You can get that from Developer Tools, tab States.

```
supported_color_modes:
  - onoff
color_mode: null
friendly_name: Dining Table Light
supported_features: 0
```

**Reply by edwardtfn** [reply]
I've just uploaded a fix to `dev`. You have to update the blueprint only, which is now beta 7, but is compatible with ESPHome and TFT beta 6.

**Reply by edwardtfn** [reply]
Please try that and let me know the results.

**Reply by edwardtfn** [reply]
Now TFT is also available on Beta 7

**Reply by lordhelmet77** [reply]
The blueprint beta 7 fixes the issues! 🥳

**Reply by hapklaar** [reply]
@edwardtfn Came here to report the same. 

Turning it off now also works again for me with the current beta blueprint, but when keeping it pressed, I still don't get the option to dim. It's now just an on off switch. 

PS the blueprint now shows ℹ️ Version: v4.2 and no longer beta
PPS attributes for the entity
```
supported_color_modes: brightness
color_mode: brightness
brightness: 136
off_with_transition: false
off_brightness: null
icon: mdi:radiator
friendly_name: Desk heater dimmer Light
supported_features: 40
```

**Reply by edwardtfn** [reply]
Let's take this in another issue:

- #1599

**Reply by edwardtfn** [reply]
@lordhelmet77, when fixing #1599 I've touched the very same area where the fix for #1584 and #1585 are so, if you have a chance it would be nice if you can give a try to the latest blueprint in `dev` and let me know the results.
If it is everything working for you I will start the release process.

**Reply by lordhelmet77** [reply]
Sure! I'll have a look in about an hour.

**Reply by edwardtfn** [reply]
## Welcome to the v4.2.1 Patch!

This patch introduces minor updates aimed at enhancing the user experience and improving documentation. Below you'll find a detailed list of the changes. While this update is designed to integrate smoothly with existing v4.2 components, we recommend updating all components to this latest version for the best performance.

### Update Instructions
For detailed update procedures, please refer to our comprehensive guide:
- [How To - Update](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/howto.md#update-blueprint)

## Overview of Changes

| Change | Criticality | Affected Components |
| :-- | :--: | :--: |
| Resolved issue with the dimming feature not activating when button is held (#1599) | Minor | `Blueprint` |
| Added "Clean Build Files" step to the update instructions for clarity (#1601) | Minor | `Documentation` |

### Key Improvements
- **Dimming Feature Fix**: Addressed an issue where the long press in the buttons was not responding as expected, ensuring smoother light control from your panel.
- **Updated Documentation**: Enhanced our update instructions to include a "Clean Build Files" step, providing clearer guidance for a more efficient update process.

Thank you for your continued support and engagement with our project. Your feedback is invaluable in helping us refine and improve the NSPanel HA Blueprint experience. For any issues or suggestions, please visit our [Issues page](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues).

**Comment by edwardtfn** [comment]
v4.2 is out, so I'm closing this discussion.

Again, thanks a lot for your support during this beta!!

Let's go to v4.3 now... 😆 

https://github.com/Blackymas/NSPanel_HA_Blueprint/releases/tag/v4.2

**Reply by hapklaar** [reply]
Great, that explains why I already saw 4.2. Did you see my last report just above?

**Reply by edwardtfn** [reply]
Just saw that... 😩

Anything new in Home Assistant logs?

**Reply by edwardtfn** [reply]
- #1599

**Comment by edwardtfn** [comment]
## Welcome to the v4.2.1 Patch!

This patch introduces minor updates aimed at enhancing the user experience and improving documentation. Below you'll find a detailed list of the changes. While this update is designed to integrate smoothly with existing v4.2 components, we recommend updating all components to this latest version for the best performance.

### Update Instructions
For detailed update procedures, please refer to our comprehensive guide:
- [How To - Update](https://github.com/Blackymas/NSPanel_HA_Blueprint/blob/main/docs/howto.md#update-blueprint)

## Overview of Changes

| Change | Criticality | Affected Components |
| :-- | :--: | :--: |
| Resolved issue with the dimming feature not activating when button is held (#1599) | Minor | `Blueprint` |
| Added "Clean Build Files" step to the update instructions for clarity (#1601) | Minor | `Documentation` |

### Key Improvements
- **Dimming Feature Fix**: Addressed an issue where the long press in the buttons was not responding as expected, ensuring smoother light control from your panel.
- **Updated Documentation**: Enhanced our update instructions to include a "Clean Build Files" step, providing clearer guidance for a more efficient update process.

Thank you for your continued support and engagement with our project. Your feedback is invaluable in helping us refine and improve the NSPanel HA Blueprint experience. For any issues or suggestions, please visit our [Issues page](https://github.com/Blackymas/NSPanel_HA_Blueprint/issues).
